misc-utils/fincore.c

```diff
@@ -537,25 +560,31 @@ int main(int argc, char ** argv)
 		}
 	}
 
-	for(; optind < argc; optind++) {
-		struct fincore_state st = {
-			.name = argv[optind],
-		};
+	if (ctl.recursive) {
+#ifdef HAVE_FTS_OPEN
+		FTS *fts = fts_open(argv + optind, FTS_PHYSICAL, NULL);
+		FTSENT *ent;
 
-		switch (fincore_name(&ctl, &st)) {
-		case 0:
-			add_output_data(&ctl, &st);
-			break;
-		case 1:
-			break; /* ignore */
-		default:
+		if (!fts) {
+			warn(_("failed to iterate tree"));
 			rc = EXIT_FAILURE;
-			break;
+		} else {
+			while ((ent = fts_read(fts)) != NULL) {
+				if (ent->fts_info == FTS_F || ent->fts_info == FTS_DEFAULT) {
+					/* fts changes directory when iterating,
+					 * so we need to use .fts_accpath to access
+					 * the file named .fts_path */
+					rc |= fincore_name(&ctl, ent->fts_accpath, ent->fts_path, ent->fts_statp);
+				}
+			}
 		}
-	}
+#endif
+	} else
+		for(; optind < argc; optind++)
+			rc |= fincore_name(&ctl, argv[optind], argv[optind], NULL);
 
```

Comment:
Multi-line code block without { } is usually hard to read. what about:

```C
} else for(; optind < argc; optind++)
      rc |= fincore_name(&ctl, argv[optind], argv[optind], NULL);
```
Or classic way with { }:
```C
} else { 
      for(; optind < argc; optind++)
               rc |= fincore_name(&ctl, argv[optind], argv[optind], NULL);
}
```

Sorry for nitpicking :-)